Shared/Samples/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesView.Model.swift

```diff
@@ -0,0 +1,273 @@
+            satelliteStatus = "Satellites info will be shown here."
+            
+            // Reset NMEA location data source.
+            nmeaLocationDataSource = nil
+            autoPanMode = .off
+
+            Task {
+                // Stop the location display, which in turn stop the data source.
+                await locationDisplay.dataSource.stop()
+            }
+            
+            // Cancel the autoPan, location, and satellite observation tasks.
+            cancelObservationTasks()
+            
+            // Pause the mock data generation.
+            mockNMEADataSource.stop()
+        }
+
+        /// Starts the location data source and awaits location and satellite updates.
+        /// - Parameter usingMockedData: Indicates that the location datasource should use mocked data.
+        func start(usingMockedData: Bool = false) {
+            // Set NMEA location data source for location display.
+            locationDisplay.dataSource = nmeaLocationDataSource
+            // Set buttons states.
+            isSourceButtonDisabled = true
+            isResetButtonDisabled = false
+            
+            // Start the data source and location display.
+            if usingMockedData {
+                mockNMEADataSource.start(with: nmeaLocationDataSource)
+            }
+            
+            // Set the autopan mode to `.recenter`
+            autoPanMode = .recenter
+
+            Task {
+                // Start the data source
+                try await locationDisplay.dataSource.start()
+            }
+            
+            // Kick off tasks to monitor autoPan, locations, and satellites.
+            observationTasks.append(
+                contentsOf: [
+                    observeAutoPanTask,
+                    observeLocationsTask,
+                    observeSatellitesTask
+                ]
+            )
+        }
+        
+        /// A detached task observing location display autoPan changes.
+        var observeAutoPanTask: Task<Void, Never> {
+            Task.detached { [unowned self] in
+                defer { print("mode in locationDisplay.$autoPanMode task ending") }
+                for await mode in locationDisplay.$autoPanMode {
+                    await MainActor.run {
+                        isRecenterButtonDisabled = nmeaLocationDataSource == nil || mode == .recenter
+                    }
+                }
+            }
+        }
+        
+        /// A detached task observing location data source location changes.
+        var observeLocationsTask: Task<Void, Never> {
+            Task.detached { [unowned self] in
+                defer { print("location in nmeaLocationDataSource.locations task ending") }
+                for await location in nmeaLocationDataSource.locations {
+                    guard let nmeaLocation = location as? NMEALocation,
+                          nmeaLocationDataSource.status == .started else { return }
```

Comment:
Why the NMEA LDS status is checked here? 🤔 

I think the LDS will stop giving location updates (from the async stream) once the LDS is stopped, or nilified.